integration-tests/utils/integration-utils.js

```diff
@@ -159,8 +159,17 @@ const select = async (page, labelContains, text) => {
   return click(page, `//div[starts-with(@id, "react-select-") and contains(normalize-space(.),"${text}")]`)
 }
 
-const waitForNoSpinners = page => {
-  return page.waitForXPath('//*[@data-icon="loadingSpinner"]', { hidden: true })
+// Used to prevent checking loadingSpinner while page is blank
+const waitUntilDomContentLoaded = page => {
+  return Promise.all([
+    page.waitForFunction(() => document),
+    page.waitForSelector('div:not(:empty):not([id="root"])')
+  ])
+}
+
+const waitForNoSpinners = async page => {
+  await waitUntilDomContentLoaded(page)
```

Comment:
It seems like `waitUntilDomContentLoaded` would only be needed on the initial page load. Since `waitForNoSpinners` is frequently used, is it worth calling `waitUntilDomContentLoaded` every time? Would putting it in `goToPage` work?